torch/fx/passes/utils/matcher_utils.py

```diff
@@ -151,15 +178,22 @@ def _remove_overlapping_matches(self, matches: List[InternalMatch]) -> List[Inte
         for match in matches:
             found_overlap = False
             for pn, gn in match.nodes_map.items():
-                if pn.op not in {"placeholder", "output"} and gn in nodes_matched:
-                    found_overlap = True
-                    break
+                if pn.op not in {"placeholder", "output"}:
+                    if isinstance(gn, list):
+                        found_overlap = any(n in nodes_matched for n in gn)
+                    else:
+                        found_overlap = gn in nodes_matched
+                    if found_overlap:
+                        break
 
             if not found_overlap:
                 non_overlapping_matches.append(match)
                 for pn, gn in match.nodes_map.items():
                     if pn.op not in {"placeholder", "output"}:
-                        nodes_matched.add(gn)
+                        if isinstance(gn, list):
+                            nodes_matched.update(gn)
```

Comment:
dont you need iterate through all nodes of the list update/add those to `nodes_matched`